switchtec.c

```diff
@@ -319,23 +319,47 @@ static ssize_t field ## _show(struct device *dev, \
 	struct device_attribute *attr, char *buf) \
 { \
 	struct switchtec_dev *stdev = to_stdev(dev); \
-	return io_string_show(buf, &stdev->mmio_sys_info->gen3.field, \
-			    sizeof(stdev->mmio_sys_info->gen3.field)); \
+	struct sys_info_regs __iomem *si = stdev->mmio_sys_info; \
+	\
+	if (stdev->gen == SWITCHTEC_GEN4) \
+		return io_string_show(buf, &si->gen4.field, \
+				      sizeof(si->gen4.field)); \
+	else \
```

Comment:
if generation equal to GEN5, it will go through GEN3 path

Reply:
It can't equal GEN5. If and when it can, we will have to address this. But we have no idea what the correct thing to do for GEN5 so there's no sense worrying about it now.